src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AmbientValueAttribute.cs

```diff
@@ -139,26 +133,14 @@ public override bool Equals(object obj)
                 return true;
             }
 
-            AmbientValueAttribute other = obj as AmbientValueAttribute;
-
-            if (other != null)
+            if (!(obj is AmbientValueAttribute other))
             {
-                if (Value != null)
-                {
-                    return Value.Equals(other.Value);
-                }
-                else
-                {
-                    return (other.Value == null);
-                }
+                return false;
             }
-            return false;
-        }
 
-        public override int GetHashCode()
-        {
-            return base.GetHashCode();
+            return Value != null ? Value.Equals(other.Value) : other.Value == null;
```

Comment:
Hmm, that looks a bit weird to me - it's subjective I guess. I've got used to seeing the actual checking at the end of the Equals method.

No biggie - I updated the code